fuel-merkle/src/sparse/merkle_tree.rs

```diff
@@ -126,12 +177,16 @@ where
     TableType: Mappable<Key = Bytes32, Value = Primitive, OwnedValue = Primitive>,
     StorageType: StorageMutate<TableType, Error = StorageError>,
 {
-    pub fn from_set<I, D>(mut storage: StorageType, set: I) -> Result<Self, StorageError>
+    pub fn from_set<B, I, D>(mut storage: StorageType, set: I) -> Result<Self, StorageError>
     where
-        I: Iterator<Item = (Bytes32, D)>,
+        I: Iterator<Item = (B, D)>,
```

Comment:
Would it be better to use `I: Iterator<Item = (MerkleTreeKey, D)>,`?

Reply:
Yes, but it will break many places, so I moved this conversion into this method. It is okay that `from_set` accepts pure keys

Reply:
We will also need to be able to bulk-insert pure keys for regenesis